modules/standard/DateTime.chpl

```diff
@@ -1197,7 +1197,36 @@ module DateTime {
     timeStruct.tm_wday = (weekday(): int(32) + 1) % 7; // shift Sunday to 0
     timeStruct.tm_yday = (this.replace(tzinfo=nil) - new datetime(year, 1, 1)).days: int(32);
 
-    strftime(c_ptrTo(buf), bufLen, fmt.c_str(), timeStruct);
+    // Iterate over format specifiers in strftime()
+    iter strftok(s)
+    {
+      var per = "";
+      for c in s {
+        if per == "" {
+          if c == '%' {
+             per += c;
+          } else {
+             yield c;
+          }
+        } else {
+          per += c;
+          if c != '-' {
```

Comment:
What format is this line looking for/trying to build? `"%-?"`?

Reply:
> What format is this line looking for/trying to build? `"%-?"`?

`strftime()` allows all numerical output specifiers to use an optional `-` modifier, which specifies that instead of 0-padding the number so that it has a constant number of digits, the number will be formatted in a platform-specific way, which in GNU libc, does not include any leading spaces or 0s as padding. So we parse for `-` after `%` as an optional modifier. We do not need to handle arbitrary widths or precisions like in `printf()`, however.

Reply:
I extended it to allow `%-*` (no padding), `%_*` (space padding), `%0*` (explicit 0 padding), and `%E*` and `%O*` (POSIX extensions), which are all allowed as modifiers by the C `strftime()`, since we are calling C `strftime()` directly.

See `man strftime` for the explanation of these modifiers.